packages/ui/src/components/organisms/Form/DynamicForm/hooks/internal/useValues/useValues.ts

```diff
@@ -26,8 +27,25 @@ export const useValues = <TValues extends object>({
     (fieldName: string, valueDestination: string, newValue: unknown) => {
       setValuesState(prev => {
         const newValues = { ...prev };
+        const parentValueDestination = valueDestination.split('.').slice(0, -1).join('.');
+
         set(newValues, valueDestination, newValue);
 
+        if (parentValueDestination) {
+          const parentValue = get(prev, parentValueDestination);
+          let newParentValue: any;
+
+          if (Array.isArray(parentValue)) {
+            newParentValue = [...parentValue];
+          }
+
+          if (typeof parentValue === 'object' && !Array.isArray(parentValue)) {
```

Comment:
Consider using `isObject` from a package or even common. Object checks also check the value is not null.